### DISCUSSION TITLE
Trying to make it easier to expose namespaces to sci

### DISCUSSION THREAD

**Discussion by phronmophobic** [borkdude/sci#615] [Show and tell]
I've created a small utility library for exposing namespaces to sci called scify, https://github.com/phronmophobic/scify.

As a bonus, it can also replace existing protocols to work with both sci values and non-sci values.

- It works even under native-image compilation with direct linking.
- Code evaluated by sci can even extend the protocols to new types defined in Sci by defrecord

See https://github.com/phronmophobic/scify/blob/main/src/com/phronemophobic/scify.clj#L5

It does rely on both clojure and sci internals, but based on https://ask.clojure.org/index.php/10967/are-protocol-methods-guaranteed-to-not-be-directly-linked, it seems like the clojure implementation details will likely be stable.

**Comment by borkdude** [comment]
@phronmophobic Thanks for sharing!
